tangelo/linq/target/target_qulacs.py

```diff
@@ -95,10 +98,15 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
                 sample = self._int_to_binstr(state.sampling(1)[0], source_circuit.width)
                 bitstr = measurement + sample
                 samples[bitstr] = samples.get(bitstr, 0) + 1
+
+                if measurement == desired_meas_result:
+                    python_statevector = state.get_vector()
+                    self._current_state = python_statevector
```

Comment:
Why not
```python
self._current_state =  state.get_vector()
```

Reply:
It's just for consistency, so the `return` statement always uses this variable.

Reply:
does this even make sense? Like, if this "current state" is not the same for all shots and thus not deterministic, what meaning does it even have ?

As a user, what do I do with that? I won't guess it unless I look at the code, and I'll understand (or not) that t's just the last state that apparently matched my desired mid-circuit measurement. May be misleading.